velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1394,4 +1399,41 @@ struct TimeZoneMinuteFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct TimestampWithTimezoneAtTimezoneFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  std::string sessionTzName = "";
+  const date::time_zone* sessionTimezone = nullptr;
+  int64_t targetTimezoneID = -1;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<TimestampWithTimezone>* /*tsWithTz*/,
+      const arg_type<Varchar>* timeZone) {
+
+    // below statements not working in init()
+    // auto timeZoneStr = std::string_view(timeZone->data(), timeZone->size());
```

Comment:
The expectation is that timeZone is a constant. Your tests make this a field.
See `TEST_F(DateTimeFunctionsTest, dateTrunc)` as an example.
You should confirm on the Java side that this argument will always be a constant.

Reply:
I don't think I quite understand what you mean that my tests make it a field. My test function `timestampWithTimezoneAtTimezone()` takes in the timezone string as a constant as well. 

Checked with Presto Java and it seems that this function does only expect the timezone arg as a constant.

Reply:
@majetideepak I don't think I quite understand what you mean that my tests make it a field. My test function `timestampWithTimezoneAtTimezone()` takes in the timezone string as a constant as well. 

Checked with Presto Java and it seems that this function does only expect the timezone arg as a constant.